modules/programs/i3lock.nix

```diff
@@ -0,0 +1,291 @@
+          };
+
+          ringvercolor = mkOption {
+            description = "i3lock ring verify color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          insidewrongcolor = mkOption {
+            description = "3lock inside wrong color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          ringwrongcolor = mkOption {
+            description = "i3lock ring wrong color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          insidecolor = mkOption {
+            description = "3lock inside color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          ringcolor = mkOption {
+            description = "i3lock ring color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          linecolor = mkOption {
+            description = "i3lock line color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          separatorcolor = mkOption {
+            description = "i3lock seprarator color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          verifcolor = mkOption {
+            description = "i3lock verify color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          wrongcolor = mkOption {
+            description = "i3lock wrong color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          timecolor = mkOption {
+            description = "i3lock time color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          datecolor = mkOption {
+            description = "i3lock date color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          layoutcolor = mkOption {
+            description = "i3lock layout color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          keyhlcolor = mkOption {
+            description = "i3lock key hold color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+
+          bshlcolor = mkOption {
+            description = "i3lock backspace hold color.";
+            type = types.nullOr types.str;
+            default = null;
+          };
+        };
+      });
+      default = null;
+    };
+
+    extraCommands = mkOption {
+      description = "Extra commands to run before running i3lock";
+      default = "";
+      type = types.lines;
+      example = ''
+        ''${pkgs.scrot}/bin/scrot /tmp/screen_locked.png
+        ''${pkgs.imagemagick}/bin/convert /tmp/screen_locked.png \
+          -scale 10% -scale 1000% /tmp/screen_locked.png
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    xsession.lockCmd =
```

Comment:
I'm probably missing something, but why we can't just enable screen-locker service and set `lockCmd` there? It was designed to be used with any locks, it just integrates them with systemd (loginctl).

```nix
services.screen-locker = {
  enbale = true;
  lockCmd = ...
};
```

Reply:
loginctl already abstracts the locking mechanism from what I read so seems like a good approach

Reply:
Agreed, thx, changed.